src/udx.c

```diff
@@ -1510,15 +1505,10 @@ process_sacks (udx_stream_t *stream, char *buf, size_t buf_len) {
 
 static void
 process_data_packet (udx_stream_t *stream, int type, uint32_t seq, char *data, ssize_t data_len) {
-  if (seq == stream->ack && type == UDX_HEADER_DATA) {
+  if (seq == stream->ack && type & UDX_HEADER_DATA) {
```

Comment:
A packet could have type DATA | END, such a packet would end up in the out-of-order queue, and then processed like normal.